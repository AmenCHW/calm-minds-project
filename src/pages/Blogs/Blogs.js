
import React, {  useState } from "react"; 
import "../../App.css";
import { collection, addDoc} from "firebase/firestore"; 
import { ReactComponent as SendIcon } from '../../icons/send.svg';
import {db} from '../../firebase-config'

function Blogs() {
  const [newEmailInput, setNewEmailInput] = useState({});
  const handleOnChange = (event) => {
    const {target: {name: keyName, value}} = event;

    console.log('handleOnChange:', keyName);

    setNewEmailInput((prev) => {
      // Copy the previous object (state) and only change the keyName that I want
      // prev is aka newMovieInput
      return { ...prev, [keyName]: value };
    });
  };

  const handleSubmit = async (event) => {
    event.preventDefault();
   
    console.log(newEmailInput);

    await addDoc(collection(db, "subscribed-emails"), {
      ...newEmailInput // { "": muslim@gmail.com } => { "email": "muslim@gmail.com" }
    });
    // Clear the form
    setNewEmailInput({
      email: "",
      
    });
  };
  return (
    <>
      {/* <EditProfile /> */}
      <div className=' mx-4 my-10'>
      <div className='py-10 px-2'>                                                                              
       <h3 className='text-xl sm:text-xl md:text-3xl  pt-5 items-start '>SIGN UP FOR THE HEALING BLOG</h3>
       <p>A weekly, ad-free Blog that helps you stay in the know.</p>
      {/* _____________________subscribtion__________________________________________________________________________________________________- */}
       <div className="mt-4 flex  sm:ml-0  justify-center">
        <form  className='flex' onSubmit={handleSubmit}>
          <input
            className="p-2 border-2 lg:px-3 px-0 border-[#718096] rounded-l-md"
            placeholder="Enter your e-mail"
            type='email'
            name='email'
            
            value={newEmailInput.email}
            onChange={handleOnChange}
          
          />
          <button
            className="p-2 border-y-2 border-r-2 border-[#718096] rounded-r-md bg-cyan-500 active:pr-1 active:pl-1 active:bg-cyan-400"
            type="submit"
            aria-label="Send"
          > <SendIcon />
          </button>
          </form>
        </div>
         {/* _____________________subscribtion___________________________________________________________________________________________________________ -*/}
        <div> 
          <h3 className='text-xl sm:text-xl md:text-3xl format-normal leading-normal pt-5 my-10'>RECOMMENED FOR YOU  </h3>
        </div>
      </div>
      </div>
      
    </>
  );
  
}

export default Blogs;
