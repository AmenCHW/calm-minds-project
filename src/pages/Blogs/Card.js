import React, { useEffect, useState } from 'react';
import { ref, listAll, getDownloadURL } from 'firebase/storage';
import { storage } from '../../firebase-config';

function Card({ title }) {
  const [imageList, setImageList] = useState([]);
  const imagesListRef = ref(storage, 'blogImages/');
  useEffect(() => {
    listAll(imagesListRef).then((response) => {
      response.items.forEach((item) => {
        getDownloadURL(item).then((url) => {
          setImageList((prev) => [...prev, url]);
        });
      });
    });
  }, []);

  return (
    <div
      className="
        flex
        justify-center
        text-3xl md:text-7xl 
        p-2 w-50 h-50  w-3/4 lg:w-60 sm:h-auto bg-blue-200
        items-center
        drop-shadow-md	
        rounded-md
        hover:shadow-lg"
    >
      <div className="flex-col">
        {imageList.map((image) => {
          return (
            <img
              src={image}
              alt=""
              className="object-cover h-48 w-96 rounded-lg"
            />
          );
        })}
        <h2 className="text-sm text-center mt-6">{title}</h2>
        <div className="text-sm hover:cursor-pointer text-gray-500" />
      </div>
    </div>
  );
}

export default Card;
