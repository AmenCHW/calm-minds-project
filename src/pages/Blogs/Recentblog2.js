import React, {  useEffect, useState} from 'react';
 import {  collection, getDocs } from 'firebase/firestore';
import { Link } from 'react-router-dom';
 import { db } from '../../firebase-config';




function RecentBlogs2() {
  const [blogs, setBlog] = useState([]);
  // ____________________________ blog id_____________________________________________________________________________________________________________________________
   const userCollectionRef = collection(db, 'blogCollection');
    useEffect(() => {
    const fetchBlogImage = async () => {
      const data = await getDocs(userCollectionRef);
      setBlog(data.docs.map((doc, i) => ({ ...doc.data(), id: doc.id, active: i<3 })));
    };

    fetchBlogImage();
  
  }, []);
  //  _________________________________________ Carsoul_____________________________________________________________________________________________________________

  const slideLeft = () => {
    const slider = document.getElementById('slider');
    slider.scrollLeft -= 380;
  };

  const slideRight = () => {
    const slider = document.getElementById('slider');
    slider.scrollLeft += 380;
  };
 
   return (
   
      
    <div className="flex flex-col md:flex-row place-content-center mb-4 mx-4">
      <button
        type="button"
        className="text-xl md:text-5xl cursor-pointer border-2 pb-2 px-5 border-black rounded-md h-1/2 my-10 md:mr-4 mx-36 md:mx-0"
        onClick={slideLeft}
      >
        &#8249;
      </button>
     
      <div    id="slider" className="flex-col flex md:flex-row gap-2 sm:gap-4 place-items-center">
      


      
         {blogs.map((blog) => {
            return (

              <div className='object-cover h-60 w-96 rounded-lg mr-3 mb-4 ' 
              key={blog.id}
               >
                <Link to= {`/blogs/${blog.id}`} onClick={window.scrollTo(0, 0)}> 
                  <img
                    src={blog.imgText}
                    alt="" 
                    className=" im1 object-cover h-56 w-96 rounded-lg mr-3 mb-4 hover:scale-110 transition duration-300 ease-in-out" 
                  />
                </Link>  

              </div>
             
          
            );

            
          })} 
         
          </div> 

      <button
        type="button"
        className="text-xl md:text-5xl hover:cursor-pointer border-2 pb-2 px-5 border-black rounded-md h-1/2 my-10 md:ml-4 mx-36 md:mx-0"
        onClick={slideRight}
      >
        &#8250;
      </button>
    
    </div>
    
   
  );
}
export default RecentBlogs2;

 