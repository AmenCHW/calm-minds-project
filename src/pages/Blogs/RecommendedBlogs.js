import React, { useEffect, useState } from 'react';
import { Link } from 'react-router-dom';
import { collection,  getDocs } from 'firebase/firestore';
// import { ref, listAll, getDownloadURL } from 'firebase/storage';
import { db } from '../../firebase-config';

function RecommendedBlogs() {
  // const [imageList, setImageList] = useState([]);
  // const imagesListRef = ref(storage, 'blogImages/');
  const [blogs, setBlog] = useState([]);
  const userCollectionRef = collection(db, 'blogCollection');
  useEffect(() => {
    const fetchBlogImage = async () => {
      const data = await getDocs(userCollectionRef);
      setBlog(data.docs.map((doc) => ({ ...doc.data(), id: doc.id })));
    };
    fetchBlogImage();
  }, []);
  // _________________showing Recommended blog images using List All and reference___________________________
  // useEffect(() => {
  //   listAll(imagesListRef).then((response) => {
  //     response.items.forEach((item) => {
  //       getDownloadURL(item).then((url) => {
  //         setImageList((prev) => [...prev, url]);
  //       });
  //     });
  //   });
  // }, []);

  
  return (
    <div className="mx-auto  px-10 py-10">
      <div className="mx-auto lg:max-w-7xl ">
          {/* _____________________Recommended blog images _________________________________________________________________________________ -*/}
          <div className="py-10 ">
          <div className="mx-2 my-3">
            <h3 className="text-2xl sm:text-2xl md:text-3xl format-normal leading-normal pt-5 mt-10 font-medium">
              RECOMMENED FOR YOU{' '}
            </h3>
            <div className="flex flex-wrap pr-5 mt-6">
              {blogs.slice(1, 3).map((blog) => {
                return (
                  <div
                    className="object-cover h-60 w-96 rounded-lg mr-3 mb-4 "
                    key={blog.id}
                  >
                   
                    <Link 
                    to= {`/blogs/${blog.id}`}
                    onClick={window.scrollTo(0, 0)}
                    >
                      <img

                        src={blog.imgText}
                        alt=""
                        className=" im1 object-cover h-56 w-96 rounded-lg mr-3 mb-4"
                      />
                    </Link>
                  </div>
                );
              })}
              {/* {imageList.slice(1, 3).map((image) => {
              return (
                <div>
               
                </div> 
              );
            })} */}
            </div>
          </div>
        </div>
      </div>
      <div>
        {/* {blog.map((blogg) => {
        return (
          <div key={blogg}>
          <div key={blogg.blogtitle}>
            <RecentBlogs imgs={blogg.coverImg}/>
            </div></div>
        );
      })} */}
      </div>
    </div>
  );
}

export default RecommendedBlogs;
