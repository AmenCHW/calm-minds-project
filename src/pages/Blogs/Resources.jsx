import { collection, getDocs } from '@firebase/firestore';
import React, { useEffect } from 'react';
// import { useState } from 'react';
import { db } from '../../firebase-config';

export default function Resources() {
//   const [blogImage, setBlogImage] = useState([]);
  const userCollectionRef = collection(db, 'blogCollection');
  useEffect(() => {
    const fetchBlogImage = async () => {
      const { data } = await getDocs.get(userCollectionRef);
      console.log(data);
    };

    fetchBlogImage();
  }, []);
  return <div>Resources</div>;
}

// const { data } = await userCollectionRef.get();
//           setBlogImage(data);
//         };
