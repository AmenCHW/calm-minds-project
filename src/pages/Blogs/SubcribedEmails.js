import React, {  useState } from 'react';
import { collection, addDoc } from 'firebase/firestore';
// import { ref, listAll, getDownloadURL } from 'firebase/storage';
import { ReactComponent as SendIcon } from '../../icons/send.svg';
import { db } from '../../firebase-config';

function SubcribedEmails() {
  const [newEmailInput, setNewEmailInput] = useState({});

  


  // __________this handling subscribtion email input___________________
  const handleOnChange = (event) => {
    const {
      target: { name: keyName, value },
    } = event;
    // console.log('handleOnChange:', keyName);
    setNewEmailInput((prev) => {
      return { ...prev, [keyName]: value };
    });
  };
  const handleSubmit = async (event) => {
    event.preventDefault();
    // console.log(newEmailInput);
    await addDoc(collection(db, 'subscribed-emails'), {
      ...newEmailInput, // { "": muslim@gmail.com } => { "email": "muslim@gmail.com" }  key value shouldnot be empty so in email input we have name=email
    });
    // Clear the form
    setNewEmailInput({
      email: '',
    });
  };

  return (
    <div className="mx-auto  px-10 py-10">
      <div className="mx-auto lg:max-w-7xl ">
        
        {/* _____________________subscribtion__________________________________________________________________________________________________- */}
        <div className="mt-4 flex mx-2 ">
          <form className="flex" onSubmit={handleSubmit}>
            <input
              className="p-2 border-2 lg:px-3 px-0 border-[#718096] rounded-l-md"
              placeholder="Enter your e-mail"
              type="email"
              name="email"
              value={newEmailInput.email}
              onChange={handleOnChange}
            />
            <button
              className="p-2 border-y-2 border-r-2 border-[#718096] rounded-r-md bg-cyan-500 active:pr-1 active:pl-1 active:bg-cyan-400"
              type="submit"
              aria-label="Send"
            >
              {' '}
              <SendIcon />
            </button>
          </form>
        </div>  
      </div>
    </div>
  );
}

export default SubcribedEmails;
