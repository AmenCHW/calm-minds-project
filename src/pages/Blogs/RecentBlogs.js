import React, { useState } from 'react';
import Card from './Card';
import { initalState } from './testData';

function RecentBlogs() {
  const [cards, setCards] = useState(initalState);

  const handleRightClick = () => {
    const prevState = [...cards];
    // find next inactive card index - top
    const nextCardIdx = prevState
      .filter((ft) => ft.active === true)
      .sort((a, b) => {
        if (a.pos > b.pos) {
          return 1;
        }
        return -1;
      })[0].idx;
    // reset
    prevState.find((f) => f.active === false).active = true;
    // update
    prevState.find((f) => f.idx === nextCardIdx).active = false;
    // maximize pos
    prevState.find((f) => f.idx === nextCardIdx).pos =
      Math.max.apply(
        null,
        prevState.map((o) => {
          return o.pos;
        })
      ) + 1;

    // update state
    setCards(prevState);
  };

  const handleLeftClick = () => {
    const prevState = [...cards];
    // find next inactive card index - bottom
    const nextCardIdx = prevState
      .filter((ft) => ft.active === true)
      .sort((a, b) => {
        if (a.pos > b.pos) {
          return 1;
        }
        return -1;
      })

      .pop(1).idx;
    // minimize pos
    prevState.find((f) => f.active === false).pos =
      Math.min.apply(
        null,
        prevState.map((o) => {
          return o.pos;
        })
      ) - 1;
    // reset
    prevState.find((f) => f.active === false).active = true;
    // update
    prevState.find((f) => f.idx === nextCardIdx).active = false;

    // update state
    setCards(prevState);
  };

  return (
    <div className="flex flex-col md:flex-row place-content-center mb-4 ">
      <button
        type="button"
        className="text-xl md:text-5xl cursor-pointer border-2 pb-2 px-5 border-black rounded-md h-1/2 my-10 md:mr-4 mx-36 md:mx-0"
        onClick={() => handleLeftClick()}
      >
        &#8249;
      </button>
      <div className="flex-col flex md:flex-row gap-2 sm:gap-4 place-items-center">
        {cards
          .filter((f) => f.active === true)
          .sort((a, b) => {
            if (a.pos > b.pos) {
              return 1;
            }
            return -1;
          })

          .map((card) => (
            <Card key={card.idx} image={card.image} title={card.text} />
          ))}
      </div>
      <button
        type="button"
        className="text-xl md:text-5xl hover:cursor-pointer border-2 pb-2 px-5 border-black rounded-md h-1/2 my-10 md:ml-4 mx-36 md:mx-0"
        onClick={() => handleRightClick()}
      >
        &#8250;
      </button>
    </div>
  );
}

export default RecentBlogs;
