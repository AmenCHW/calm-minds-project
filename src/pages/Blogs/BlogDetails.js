import React, { useEffect, useState } from 'react';
import { getDoc, doc } from 'firebase/firestore';
import { useParams } from 'react-router-dom';
import { db } from '../../firebase-config';
import SubcribedEmails from "./SubcribedEmails"

const BlogDetails = () => {
  const params = useParams({});
  const [blog, setBlog] = useState([]);
  useEffect(() => {
    async function fetchListing() {
      const docRef = doc(db, 'blogCollection', params.id);
      const docSnap = await getDoc(docRef);

      setBlog(docSnap.data());
    }
    fetchListing();
  }, [params.listingId]);

  return (
    <div className="mx-auto px-10 py-10">
      <div className="mx-auto lg:max-w-7xl">
        <div className="rounded-lg bg-cover bg-center ">
          <div className="rounded-lg bg-cover bg-center ">
            <img
              src={blog.coverImg}
              alt=""
              className="object-cover h-full w-full object-center  rounded-lg mb-20"
            />
          </div>
          <h1 className="text-6xl font-medium mb-12 mt-4">{blog.blogtitle}</h1>
          <p className="text-2xl">{blog.summery}</p>
          <h3 className="text-5xl mb-6 mt-12 ">{blog.secondTitle}</h3>
          <p className="text-2xl leading-relaxed">{blog.paragraph}</p>
        </div> 
      </div>
      <SubcribedEmails/>
    </div>
  );
};

export default BlogDetails;


// blogsCollection/${id}
