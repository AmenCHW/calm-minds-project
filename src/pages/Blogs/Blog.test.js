import React from "react";
 import {render,screen } from "@testing-library/react"
import Blogs from "./Blogs";
import renderer from 'react-test-renderer' 

test("first snapshot testing",()=>{
    const component=renderer.create( <Blogs/>).toJSON()  
    expect( component).toMatchSnapshot();
})



test("test healing", async() => {
    render(
       <Blogs title ="Healing" />
   );
 const headingElement = screen.getAllByText(/healing/i);
   expect(headingElement).toBeInTheDocument;
 });
 
 
//  it("test <p> ", async() => {
//      render(
//         <About paragraph ="some cool one liner!" />
//     );
//     const headingElement = screen.getByText(/About/i);  //case sensittive?
//     expect(headingElement).toBeInTheDocument();
//    });