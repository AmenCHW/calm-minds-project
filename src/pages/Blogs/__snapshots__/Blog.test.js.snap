// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`first snapshot testing 1`] = `
<div
  className="mx-auto  px-10 py-10"
>
  <div
    className="mx-auto lg:max-w-7xl "
  >
    <div
      className="rounded-lg bg-cover bg-center "
    >
      <img
        alt=""
        className="object-cover h-full w-full object-center  rounded-lg mb-20"
        src="https://firebasestorage.googleapis.com/v0/b/calm-minds-project.appspot.com/o/blogImages%2FRectangle45.svg?alt=media&token=9eaa7a99-680b-4734-868c-cdf5899133e9"
      />
    </div>
    <div />
    <div
      className="mt-3 "
    >
      <h3
        className="text-xl sm:text-xl md:text-2xl  mb-2 mt-8 items-start font-medium"
      >
        SIGN UP FOR THE HEALING BLOG
      </h3>
      <p>
        A weekly, ad-free Blog that helps you stay in the know.
      </p>
      <div
        className="mx-auto  px-10 py-10"
      >
        <div
          className="mx-auto lg:max-w-7xl "
        >
          <div
            className=" flex mr-4 "
          >
            <form
              className="flex"
              onSubmit={[Function]}
            >
              <input
                className=" border-2 lg:px-3 px-0 border-[#718096] rounded-l-md"
                name="email"
                onChange={[Function]}
                placeholder="Enter your e-mail"
                type="email"
              />
              <button
                aria-label="Send"
                className="p-2 border-y-2 border-r-2 border-[#718096] rounded-r-md bg-cyan-500 active:pr-1 active:pl-1 active:bg-cyan-400"
                type="submit"
              >
                 
                <svg>
                  send.svg
                </svg>
              </button>
            </form>
          </div>
        </div>
      </div>
    </div>
    <div
      className="py-10 "
    >
      <div
        className="mx-2 my-3"
      >
        <h3
          className="text-2xl sm:text-2xl md:text-3xl format-normal leading-normal pt-5 mt-10 font-medium"
        >
          RECOMMENED FOR YOU
           
        </h3>
        <div
          className="flex flex-wrap pr-5 mt-6"
        />
      </div>
    </div>
  </div>
  <div />
</div>
`;
